Dockerfile

```
@@ -4,92 +4,81 @@ FROM debian:latest
-RUN update-exim4.conf && service exim4 restart
-
-# Set up users and permissions
-RUN adduser --gecos '' --uid 1000 --gid 50 --disabled-password php \
+RUN >&2 echo "# Setting up users and permissions..." \
+    && adduser --gecos '' --uid 1000 --gid 50 --disabled-password php \
     && adduser php staff
 
-# Install PHP 7 and some common extensions
-RUN wget -O - https://www.dotdeb.org/dotdeb.gpg | apt-key add -
-RUN echo "deb https://packages.dotdeb.org $(lsb_release -sc) all" | tee /etc/apt/sources.list.d/php.list
-RUN echo "deb-src https://packages.dotdeb.org $(lsb_release -sc) all" | tee -a /etc/apt/sources.list.d/php.list
-RUN apt-get -y update \
+RUN >&2 echo "# Install PHP 7 with extensions and update settings" \
+    && wget -O - https://www.dotdeb.org/dotdeb.gpg | apt-key add - \
+    && echo "deb https://packages.dotdeb.org $(lsb_release -sc) all" | tee /etc/apt/sources.list.d/php.list \
+    && echo "deb-src https://packages.dotdeb.org $(lsb_release -sc) all" | tee -a /etc/apt/sources.list.d/php.list \
+    && apt-get -y update \
     && apt-get install -y php7.0-fpm php7.0-cli php7.0-mbstring php7.0-mcrypt php7.0-intl \
-    php7.0-mysql php7.0-redis php7.0-memcached php7.0-gd php7.0-curl php7.0-xsl
-# Update some PHP settings
-RUN sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/cli/php.ini
-RUN sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/fpm/php.ini
-# Composer updates can require a lot of memory
-RUN sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 4G/" /etc/php/7.0/cli/php.ini
-RUN sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 256M/" /etc/php/7.0/fpm/php.ini
-RUN phpenmod -v 7.0 -s ALL mbstring mcrypt intl pdo_mysql redis memcached gd curl xml xsl
+    php7.0-mysql php7.0-redis php7.0-memcached php7.0-gd php7.0-curl php7.0-xsl php-imagick \
+    && sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/cli/php.ini \
+    && sed -i "s/^;\(date\.timezone\(\s*\)\?=\).*/\1 Europe\/Amsterdam/" /etc/php/7.0/fpm/php.ini \
+    && sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 4G/" /etc/php/7.0/cli/php.ini \
+    && sed -i "s/^\(memory_limit\(\s*\)\?=\).*/\1 256M/" /etc/php/7.0/fpm/php.ini \
+    && phpenmod -v 7.0 -s ALL mbstring mcrypt intl pdo_mysql redis memcached gd curl xml xsl imagick
 
 # Copy in php-fpm config file
 COPY php-fpm.conf /etc/php/7.0/fpm/php-fpm.conf
 
-# Install composer
-RUN wget -O - https://getcomposer.org/installer | php \
+RUN >&2 echo "# Installing tooling (composer, nodejs)..." \
+    && EXPECTED_SIGNATURE=$(wget -q -O - 'https://composer.github.io/installer.sig') \
+    && wget -O composer-installer.php 'https://getcomposer.org/installer' \
+    && INSTALLER_SIGNATURE=$(openssl dgst --sha384 -hex composer-installer.php | sed 's/^.* \(.*\)$/\1/') \
+    && if [ "$INSTALLER_SIGNATURE" != "$EXPECTED_SIGNATURE" ] ; then >&2 echo 'ERROR: Invalid installer signature' && exit 1 ; fi \
+    && php composer-installer.php \
```

Can be done in one line
```suggestion
    && php composer-installer.php --install-dir=/usr/local/bin --filename=composer \
```

---

Done.